### DISCUSSION TITLE
How to include external CSS files in Windi's compilation?

### DISCUSSION THREAD

**Discussion by iocouto** [windicss/windicss#531] [Q&A]
I'm trying to find instructions on how to 'include' an external CSS file in Windi's compilation process - that is, how to include my own custom CSS styles along with the CSS extracted by WindiCSS from my templates. This process is described clearly [in the Tailwind docs](https://tailwindcss.com/docs/installation#using-a-custom-css-file), but nowhere in Windi's.

WindiCSS docs tells us that we can [use @apply](https://windicss.org/features/directives.html) and other directives, but how/where?... If we put it in an external CSS file - as we can do with Tailwind - then how do we get WindiCSS to include this CSS file when processing our styles? 

Any guidance would be greatly appreciated.

**Comment by antfu** [comment]
I think it should work by listing your file under `extract.include`

https://windicss.org/guide/extractions.html#scanning

**Reply by iocouto** [reply]
Thanks for the suggestion! Would you have a config example that a newbie could use?... ;-)

**Comment by blackawn** [comment]


**Reply by iocouto** [reply]
@blackawn unfortunately, I haven't been able to work out how to use my own external CSS file with WindiCSS, if processing directly with the CLI. 

[In the docs](https://windicss.org/integrations/cli.html#style-block), I can see that we should be able to use the `@apply` and other directives inside a `<style>` block element in the HTML page itself, and then have Windi parse the block by using the `--style` flag in the CLI — but no info on whether we can put these directives in an external CSS file...

**Reply by ghost** [reply]
Yes, it doesn’t seem to parse the css file, it feels very unsuitable to use the <style lang="windi"> tag block

**Reply by pedromelo222** [reply]
I just found a workaround, use comment to set style inside .css
`menu.css`
```
//<style lang="windi">
.w-menu {
    @apply bg-base-100 w-56 shadow shadow-md select-none;
  }
//</style>
```

**Comment by carere** [comment]
I tried the above workaround, but I don't have any hover capabilities, and i got weird syntactic coloration for CSS file.

Is there no way to juste include a simple CSS file to the compilation process ???
And also be able to get the same feautre that provides Windi Vscode extension for TSX file in CSS file ???

Is WindiCSS only applicable to Vue ?? With only way to define custome CSS, is with Single File Component ????? (I'm using SolidJS)
